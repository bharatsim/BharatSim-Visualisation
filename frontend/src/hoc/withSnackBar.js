import React from 'react';
import { SnackbarProvider } from 'notistack';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import failureIcon from '../assets/images/error-icon.svg';
import successIcon from '../assets/images/success-icon.svg';

const useSnackBarStyles = makeStyles((theme) => ({
  contentRoot: {
    padding: theme.spacing(1, 4),
  },
  message: {
    padding: 0,
  },
  variantSuccess: {
    backgroundColor: `#EAF4E5 !important`,
    borderColor: `${theme.palette.success.main}33`,
    border: '1px solid',
    color: `${theme.palette.success.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
  variantError: {
    backgroundColor: `#FBE9E4 !important`,
    borderColor: `${theme.palette.error.light}33`,
    border: '1px solid',
    color: `${theme.palette.error.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
}));

const useIconStyles = makeStyles((theme) => ({
  iconRoot: {
    marginRight: theme.spacing(2),
    height: theme.spacing(10),
    width: theme.spacing(10),
  },
}));

function withSnackBar(WrappedComponent) {
  return (props) => {
    const snackBarClasses = useSnackBarStyles();
    const iconClasses = useIconStyles();
    const snackBarRef = React.createRef();
    const onClickDismiss = (key) => () => {
      snackBarRef.current.closeSnackbar(key);
    };
    return (
      <SnackbarProvider
        maxSnack={3}
        ref={snackBarRef}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        classes={snackBarClasses}
        iconVariant={{
          error: <img src={failureIcon} alt="error" className={iconClasses.iconRoot} />,
          success: <img src={successIcon} alt="success" className={iconClasses.iconRoot} />,
        }}
        action={(key) => <Button onClick={onClickDismiss(key)}>Dismiss</Button>}
      >
        <WrappedComponent {...props} />
      </SnackbarProvider>
    );
  };
}

export default withSnackBar;
