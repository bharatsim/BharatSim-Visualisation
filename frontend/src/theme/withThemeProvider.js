import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
