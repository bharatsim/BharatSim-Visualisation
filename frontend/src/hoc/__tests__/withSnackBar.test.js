import React, { useEffect } from 'react';
import { useSnackbar } from 'notistack';
import { render } from '@testing-library/react';
import withSnackBar from '../withSnackBar';
import withThemeProvider from '../../theme/withThemeProvider';

function DummyComponent() {
  const { enqueueSnackbar } = useSnackbar();
  useEffect(() => {
    enqueueSnackbar('test snack bar');
  });
  return <div>DummyController </div>;
}

describe('withSnackBar', () => {
  const Component = withThemeProvider(withSnackBar(DummyComponent));
  it('should provide snackbar context to child component', () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
