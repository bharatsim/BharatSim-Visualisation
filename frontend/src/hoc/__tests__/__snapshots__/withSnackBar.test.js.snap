// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSnackBar should provide snackbar context to child component 1`] = `
<div>
  <div>
    DummyController 
  </div>
  <div
    class="makeStyles-root-6 makeStyles-bottom-10 makeStyles-right-14 makeStyles-bottom-10 makeStyles-right-14 makeStyles-reverseColumns-7"
  >
    <div
      class="MuiCollapse-container SnackbarItem-collapseContainer-33 MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper SnackbarItem-collapseWrapper-34"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiSnackbar-root SnackbarItem-root-17 SnackbarItem-wrappedRoot-32 MuiSnackbar-anchorOriginBottomRight SnackbarItem-anchorOriginBottomRight-21"
          >
            <div
              aria-describedby="client-snackbar"
              class="ForwardRef-root-36 SnackbarItem-contentRoot-29 makeStyles-contentRoot-1"
              role="alert"
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <div
                class="SnackbarItem-message-30 makeStyles-message-2"
                id="client-snackbar"
              >
                test snack bar
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
